include/sys/zfs_sa.h

```diff
@@ -87,6 +88,8 @@ typedef enum zpl_attr {
 #define	SA_UID_OFFSET		24
 #define	SA_GID_OFFSET		32
 #define	SA_PARENT_OFFSET	40
+#define	SA_FLAGS_OFFSET		48
+#define	SA_PROJID_OFFSET	128
```

Comment:
Seems good, but maybe more affect on the existing files for upgrade case, is it OK? Then can I move it more close to UID/GID, such as just after flags attribute?